ipfs/defaults.go

```diff
@@ -0,0 +1,201 @@
+		Bootstrap: ipfscfg.BootstrapPeerStrings(BootstrapPeers),
+		// Configure all the addresses the node listens to
+		Addresses: ipfscfg.Addresses{
+			// List of libp2p addresses to listen to
+			Swarm: []string{
+				// listen on all network interfaces.
+				// NOTE: We don't use IPv6 for now, but we might add support for it at some point.
+				"/ip4/0.0.0.0/tcp/4001",
+			},
+			API: []string{
+				"/ip4/127.0.0.1/tcp/5001",
+			},
+			// List of public libp2p WAN address the Node can be connected with
+			// NOTE: Should be configured by Node administrator manually
+			Announce: nil,
+			// List of libp2p address the Node listens to but not accessible from the WAN.
+			// It's a good manner to fil this, as that preserves connected peers from trying connections to those
+			// NOTE: Should be configured by Node administrator manually
+			NoAnnounce: nil,
+			// List of addresses IPFS API server listens to
+			// we do not run IPFS Gateway
+			Gateway: nil,
+		},
+		// Networking configuration
+		Swarm: ipfscfg.SwarmConfig{
+			// auto port forwarding is nice
+			DisableNatPortMap: false,
+			// metrics are useful
+			DisableBandwidthMetrics: false,
+			// relays are cool from engineering standpoint, but don't use them
+			EnableRelayHop: false, EnableAutoRelay: false,
+			// ConnMgr manager controls the amount of conns a Node holds. If the amount of conns hits HighWater, ConnMgr
+			// trims amount of connections down to LowWater
+			// NOTE: Connections can by prioritized using Tagging in ConnMgr in order to keep important conns alive
+			ConnMgr: ipfscfg.ConnMgr{
+				// we use defaults from IPFS for HighWater
+				HighWater: 900,
+				// we use defaults from IPFS for HighWater
+				LowWater: 600,
+				// defines how often ConnMgr should check conns for trimming, we use default from IPFS
+				GracePeriod: "20s",
+				// just a basic connection manager
+				Type: "basic",
+			},
+			// Configuration for all networking protocols
+			Transports: ipfscfg.Transports{
+				Network: transport{
+					// we default to TCP as it's most common, battle tested transport, but we may migrate to the one below
+					TCP: ipfscfg.False,
+					// we might default to QUIC at some point, but for now TCP is more common and reliable
+					QUIC: ipfscfg.False,
+					// we don't use Websocket as we don't target for browser nodes
+					Websocket: ipfscfg.False,
+					// we don't use relays as it's expensive in terms of traffic for relay nodes
+					Relay: ipfscfg.False,
+				},
+				// Security protocol wraps IO conns in a private encrypted tunnel
+				Security: security{
+					// we default to Noise, as it's designed for p2p interactions in the first place and is 0-RTT
+					Noise: ipfscfg.DefaultPriority,
+					// we don't use SECIO as it's now deprecated and has know issues
+					SECIO: ipfscfg.Disabled,
+					// we don't allow TLS option, even it's a production ready protocol, to lower bug vectors
```

Comment:
```suggestion
					// we don't allow TLS option, despite being a production ready protocol, to lower bug vectors
```

What are bug vectors? Attack vectors?

Reply:
I didn't come up with a proper explanation. I just don't want to add two security options and just rely on one. Relying on both increases amount of code paths where things might go wrong thus bug vectors.

Reply:
Maybe change the comment to simply: 
we don't allow TLS option as we use noise only 🤷🏼